pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	if d.Output == "dot" {
+		return dotDump(g, stream, tag)
+	}
+	return g.String(), nil
+}
+
+// parseInput pasrses user input and returns a stream name, a tag
+// and in case of empty input a true condition meaning the caller
+// should look for all imageStreams in the current namespace
+// TODO: Move this under build-chain validation
+func parseInput(input string) (string, string, bool, error) {
+	args := strings.Split(input, ":")
+	if len(args[0]) == 0 {
+		return "", "", true, nil
+	}
+
+	var name, tag string
+	switch len(args) {
+	case 1:
+		name = args[0]
+		tag = imageapi.DefaultImageTag
+	case 2:
+		name = args[0]
+		tag = args[1]
+	default:
+		return "", "", false, fmt.Errorf("invalid stream:tag combination %q", input)
+	}
+	return name, tag, false, nil
+}
+
+// isFromBuildConfigLoader loads image streams found in image change triggers
+// of all build configurations in the provided namespace
+type isFromBuildConfigLoader struct {
+	namespace string
+	lister    client.ImageStreamsNamespacer
+	bc        *bcLoader
+	items     []imageapi.ImageStream
+}
+
+// Load image streams found in image change triggers of
+// all build configurations in the provided namespace
+func (l *isFromBuildConfigLoader) Load() error {
+	err := l.bc.Load()
+	if err != nil {
+		return err
+	}
+	return l.loadStreams()
+}
+
+// AddToGraph adds all the loaded image streams and their tags
+// in the given graph
+func (l *isFromBuildConfigLoader) AddToGraph(g osgraph.Graph) error {
+	for i := range l.items {
+		imagegraph.EnsureImageStreamNode(g, &l.items[i])
+		imagegraph.EnsureAllImageStreamTagNodes(g, &l.items[i])
+	}
+
+	return nil
+}
+
+// loadStreams iterates over a given set of build configurations
+// and extracts all the image streams which trigger a build when
+// the image stream is updated
+func (l *isFromBuildConfigLoader) loadStreams() error {
+	avoidDuplicates := map[*imageapi.ImageStream]struct{}{}
+	for _, cfg := range l.bc.items {
+		glog.V(1).Infof("Scanning BuildConfig %q", cfg.Name)
+		for _, tr := range cfg.Spec.Triggers {
+			from := buildutil.GetImageStreamForStrategy(cfg.Spec.Strategy)
+			if tr.ImageChange != nil && from != nil && from.Name != "" {
+				glog.V(1).Infof("found ICT with from %s kind %s", from.Name, from.Kind)
+				switch from.Kind {
+				case "ImageStreamTag":
+					bits := strings.Split(from.Name, ":")
+					stream, err := l.lister.ImageStreams(l.namespace).Get(bits[0])
+					if err != nil {
+						return err
+					}
+					avoidDuplicates[stream] = struct{}{}
+				default:
+					// ImageStreamImage and DockerImage are never updated and so never
+					// trigger builds.
+					continue
+				}
+
+			}
+		}
+	}
+
+	for stream := range avoidDuplicates {
+		l.items = append(l.items, *stream)
+	}
+
+	return nil
+}
+
+// dotDump traverses the given graph starting from the provided stream node
+// and returns the stream's dependency tree in DOT syntax
+func dotDump(g osgraph.Graph, stream graph.Node, tag string) (string, error) {
```

Comment:
> If I remove all the inbound edges to the ist of interest then essentially the subgraph we are interested in is already partitioned right?

No.  Since you have all the bcs, you could have many bc islands that you don't need.